components/org.wso2.carbon.identity.sso.agent/src/main/java/org/wso2/carbon/identity/sso/agent/bean/SSOAgentConfig.java

```diff
@@ -192,8 +226,80 @@ public KeyStore getKeyStore() throws org.wso2.carbon.identity.sso.agent.exceptio
     public void setKeyStore(KeyStore keyStore) {
         this.keyStore = keyStore;
     }
-    public void initConfig(Properties properties) throws SSOAgentException {
 
+    public void initConfig(Properties properties)
+            throws SSOAgentException, NoSuchPaddingException, InvalidKeyException, NoSuchAlgorithmException,
+            IllegalBlockSizeException, BadPaddingException, InvalidAlgorithmParameterException {
+
+        byte[] key = null;
+        String decodedPassword;
+        boolean isReadFile = false;
+        Properties copyOfProperties = new Properties();
+        copyOfProperties.putAll(properties);
+        Path path;
+        URL url = getClass().getClassLoader().getResource(SSOAgentConstants.SSOAgentConfig.SECRETKEY_FILEPATH);
+        for (Map.Entry<Object, Object> entry : copyOfProperties.entrySet()) {
+            if (String.valueOf(entry.getValue()).startsWith("Enc:")) {
+                if (System.getProperty(SECRET_KEY) == null) {
+                    if (!isReadFile) {
```

Comment:
Is there any reason for multiple if conditions?
Can't it be aggregated with AND operator? It would look more cleaner.